### DISCUSSION TITLE
Performance Tuning

### DISCUSSION THREAD

**Discussion by RussellAult** [LizardByte/community#63] [Sunshine] [answered]
So far I haven't seen anything really definitive about tuning the performance of Sunshine, so I thought I'd start the ball rolling and see what everyone else is doing.

My use-case: in-home game streaming, including first-person shooters
My goal: lowest possible round-trip latency without obvious visual sacrifices (e.g. screen tearing)

My current settings:
- On the host:
  - Hardware encoding
  - HEVC disabled
  - V-Sync disabled (in-game)
- On the guest:
  - Wired network connection
  - 1080p60 at 10 Mbit/s
  - V-Sync enabled

Things I haven't played around with much yet:
- On the host:
  - Encoder tweaks beyond disabling HEVC (What I'll try next; how did I miss these?!)
  - Running Sunshine as Administrator (Does GPU priority impact Sunshine's performance much?)
- On the guest
  - Trying more bitrates (Is lower actually faster? Do hardware encoders have a bitrate sweet-spot?)
  - Frame pacing (Won't this just slow things down?)

So far I'm reasonably happy with the setup's performance, although I feel like there might still be room for improvement. Does Sunshine have a function that will spit out realtime encoding latency statistics (like how Moonlight will display decoding latency statistics)?

**Comment by ReenigneArcher** [comment, marked as answer]
I know this post is old, but in the process of migrating discussions I figured I would share something new.

New docs have a performance tuning section: https://docs.lizardbyte.dev/projects/sunshine/en/master/md_docs_2performance__tuning.html

Feel free to submit any tips/tricks as a PR.

**Comment by willow-sprite** [comment]
this might not be useful here, but for a radius of ~2 rooms, using my pc's hotspot rather than my modem for moonlight on android is faster.

I think it was around 20ms latency shown in moonlight stats, it's been a while since I checked.

note: my modem is an old isp provided one, so this probably doesn't apply to a modern riced up modem or jeff geerling style pi router config.

[a comment by an automated account is not shown]
